### PR TITLE
added http api for deleting auth roles

### PR DESCRIPTION
**Pull request by invisig0th**

### CODE REVIEW

File: synapse/lib/httpapi.py [hunk 1]
```diff
@@ -280,7 +280,19 @@ class AuthUsersV1(Handler):
         if not await self.reqAuthUser():
             return
 
-        self.sendRestRetn([u.pack() for u in self.cell.auth.users()])
+        try:
+
+            archived = int(self.get_argument('archived', default='0'))
+
+        except Exception as e:
+            return self.sendRestErr('BadHttpParam', 'The parameter "archived" must be 0 or 1 if specified.')
```
Comment (vEpiphyte) [inline synapse/lib/httpapi.py]: This actually allows values outside of `(0, 1)` to be passed in via API.

File: synapse/lib/httpapi.py [hunk 2]
```diff
@@ -513,6 +529,31 @@ class AuthAddRoleV1(Handler):
         self.sendRestRetn(role.pack())
         return
 
+class AuthDelRoleV1(Handler):
+
+    async def post(self):
+
+        if not await self.reqAuthAdmin():
+            return
```
Comment (vEpiphyte) [inline synapse/lib/httpapi.py]: should we send a resterror here?

File: synapse/lib/httpapi.py [hunk 3]
```diff
@@ -513,6 +529,31 @@ class AuthAddRoleV1(Handler):
...
+
+        body = self.getJsonBody()
+        if body is None:
+            return
```
Comment (vEpiphyte) [inline synapse/lib/httpapi.py]: send an error here?

File: synapse/tests/utils.py [hunk 4]
```diff
@@ -989,10 +989,24 @@ def getAsyncLoggerStream(self, logname, mesg=''):
             slogger.removeHandler(handler)
 
     @contextlib.asynccontextmanager
-    async def getHttpSess(self):
+    async def getHttpSess(self, auth=None, port=None):
+
```
Comment (vEpiphyte) [inline synapse/tests/utils.py]: should allow us to override the auth data here

File: synapse/lib/httpapi.py (hunk 2 above)
Comment (invisig0th) [reply synapse/lib/httpapi.py]: The reqAuthAdmin API sends one ( similar to getJsonBody )

File: synapse/lib/httpapi.py (hunk 3 above)
Comment (invisig0th) [reply synapse/lib/httpapi.py]: it does :)

File: synapse/lib/httpapi.py (hunk 1 above)
Comment (invisig0th) [reply synapse/lib/httpapi.py]: added a check

**Pull request merged**